From: "Chen, Gong" <gong.chen@linux.intel.com>
Date: Fri, 18 Oct 2013 14:29:25 -0700
Subject: DMI: Parse memory device (type 17) in SMBIOS
Patch-mainline: Never, SUSE-Xen specific
References: fate#315508

This patch adds a new interface to decode memory device (type 17)
to help error reporting on DIMMs.

Original-author: Tony Luck <tony.luck@intel.com>
Signed-off-by: Chen, Gong <gong.chen@linux.intel.com>
Acked-by: Naveen N. Rao <naveen.n.rao@linux.vnet.ibm.com>
Acked-by: Borislav Petkov <bp@suse.de>
Reviewed-by: Mauro Carvalho Chehab <m.chehab@samsung.com>
Signed-off-by: Tony Luck <tony.luck@intel.com>
Automatically created from "patches.arch/05-dmi-parse-memory-device-type-17-in-smbios.patch" by xen-port-patches.py

--- a/arch/x86/kernel/setup-xen.c
+++ b/arch/x86/kernel/setup-xen.c
@@ -1119,6 +1119,7 @@ void __init setup_arch(char **cmdline_p)
 
 	if (is_initial_xendomain()) {
 		dmi_scan_machine();
+		dmi_memdev_walk();
 		dmi_set_dump_stack_arch_desc();
 	} else {
 		int ver = HYPERVISOR_xen_version(XENVER_version, NULL);
